### PR TITLE
updated lmerge

### PR DESCRIPTION
**Pull request by abelhj**

### CODE REVIEW

File: svtools/lmerge.py [hunk 1]
```diff
@@ -240,280 +84,410 @@ def merge(BP, sample_order, v_id, use_product, include_genotypes=False):
...
+        for i in range(len(a_L[c_i])):
+            p_L[i] += a_L[c_i][i]
+
+        for i in range(len(a_R[c_i])):
+            p_R[i] += a_R[c_i][i]
 
-            for i in range(len(a_R[c_i])):
-                p_R[i] += a_R[c_i][i]
+    #sys.exit(1)
+    ALG = 'SUM'
```
Comment (ernfrid) [inline svtools/lmerge.py]: Would be nice to remove the commented out lines from this function before merging.

File: svtools/lmerge.py [hunk 2]
```diff
@@ -240,280 +84,410 @@ def merge(BP, sample_order, v_id, use_product, include_genotypes=False):
...
+    else:
+        var.set_info('END', new_pos_R )
+    
+    var.set_info('CIPOS95', cipos95)
+    var.set_info('CIEND95', ciend95)
+    var.set_info('CIPOS', ','.join([str(x) for x in [-1*max_i_L, len(p_L) - max_i_L - 1]]))
+    var.set_info('CIEND', ','.join([str(x) for x in [-1*max_i_R, len(p_R) - max_i_R - 1]]))
+    var.set_info('PRPOS', ','.join([str(x) for x in p_L]))
+    var.set_info('PREND', ','.join([str(x) for x in p_R]))
+    #var.set_info('IMPRECISE', True)
```
Comment (ernfrid) [inline svtools/lmerge.py]: Probably should remove this commented out line as well.

File: svtools/lmerge.py [hunk 3]
```diff
@@ -530,80 +504,82 @@ def r_cluster(BP_l, sample_order, v_id, use_product, include_genotypes=False):
...
+    
+    v_id = 0
+
+    in_header = True
+    header = []
+    vcf = Vcf()
+    vcf_out=sys.stdout
+    if weight_by_samples and weight_by_support:
+        sys.stderr('Error:  Can only weight by total support OR carrier count, but not both')
+        sys.exit(1)
```
Comment (ernfrid) [inline svtools/lmerge.py]: This check can be removed if the options for these are changed to be 'choice' style. If we weren't to do this, I'd advocate for placing this check directly after command line parsing.

File: tests/lmerge_tests.py [hunk 4]
```diff
@@ -15,7 +15,8 @@ def run_integration_test(self):
         test_directory = os.path.dirname(os.path.abspath(__file__))
         self.test_data_dir = os.path.join(test_directory, 'test_data', 'lmerge')
         input_file = os.path.join(self.test_data_dir, 'input')
-        expected_result = os.path.join(self.test_data_dir, 'expected.vcf')
+        #expected_result = os.path.join(self.test_data_dir, 'expected.vcf')
```
Comment (ernfrid) [inline tests/lmerge_tests.py]: Should remove this commented out line.

Comment (ernfrid) [inline tests/lmerge_tests.py]: Probably also the old data file.

Comment (ernfrid) [inline tests/lmerge_tests.py]: Two more questions:

1. We don't have a test of including the genotypes. Has that been tested at all with the new code?
2. Seems like we're adding a format field to the output now. I assume that doesn't matter, but I think that makes the output an invalid VCF.

File: svtools/lmerge.py [hunk 5]
```diff
@@ -240,280 +84,410 @@ def merge(BP, sample_order, v_id, use_product, include_genotypes=False):
...
+        if not include_genotypes: 
+            varstring='\t'.join(varstring.split('\t', 10)[:9])
+
+        vcf_out.write(varstring+'\n')
+
+
+    else:
+        varstring=var.get_var_string(use_cached_gt_string=True)
+        if not include_genotypes: 
+            varstring='\t'.join(varstring.split('\t', 10)[:9])
```
Comment (ernfrid) [inline svtools/lmerge.py]: If we're not including genotypes then we probably don't want to include the format column. Looking at the current VCF code - the hacky way to do this would be to set var.format_dict to None. 

That's probably inappropriate though for the long term. I suspect the real change should be in the VCF classes. Perhaps by adding a test if the gts are None and if so, not printing out format.
